File: src/main/java/com/extension/factcheck/config/RestClientConfig.java
package com.extension.factcheck.config;

import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestClient;

@Slf4j
@Configuration
public class RestClientConfig {

  @Bean
  public RestClient restClient() {
    return RestClient.builder()
        .defaultStatusHandler(
            statusCode -> !statusCode.is2xxSuccessful(),
            (request, response) -> {
              log.error("API 응답 오류: " + response);
            }
        )
        .build();
  }

}

File: src/main/kotlin/com/extension/factcheck/config/RestClientConfig.kt
package com.extension.factcheck.config

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@Configuration
class RestClientConfig {

    private val log: Logger = LoggerFactory.getLogger(javaClass.simpleName)

    @Bean
    fun restClient(): RestClient {
        return RestClient.builder()
            .defaultStatusHandler(
                { statusCode -> !statusCode.is2xxSuccessful },
                { request, response ->
                    log.error("API 응답 오류: {}", response)
                },
            )
            .build()
    }
}
